src/LibDeclarations/NativeExtensions/FeaturePoints.md

```diff
@@ -0,0 +1,69 @@
+# Feature point cloud
+## Introduction
+This document proposes an API for exposing Feature Point Clouds for WebXR.
+
+## Overview
+The API is synchronous, and on demand.  When served a new XRFrame the consuming app may query for the current feature point cloud for a given frame. As feature point clouds are comprised of a large set of raw data it is best not to expose this to the JS application unless needed for app processing, so feature points must be requested in an on-demand fashion in order to reduce the amount of data presented to the consuming app, which must set its own frequency for updating its world understanding.
+
+The returned array will contain the feature point cloud tracked in the current frame, including points that may not be currently visible. Feature points will be presented in a flat array of numbers where each point is presented as a set of 5 entries representing the X, Y, Z position in world space, the confidence value, and an ID that is durable across frames.
+
+## Feature Points - API Changes
+The below details the changes to the existing API to allow querying for feature points.
+
+XRFrame will expose a new attribute featurePointCloud which returns a set that includes all feature points present in the given frame:
+```typescript
+interface XRSession {
+  ...
+  trySetFeaturePointCloudEnabled(enabled: boolean): boolean;

```

Comment:
should that return a promise? are there no async-porcesses when enabling it?

Reply:
Right now it does not do anything async it just tells the session to start collecting them similar to the updateWorldTrackingState call for plane detection.